core/execution/engine.go

```diff
@@ -1330,9 +1330,20 @@ func (e *Engine) OnTick(ctx context.Context, t time.Time) {
 		}
 		e.removeMarket(id)
 	}
+
+	// sort the marketCPStates by ID since the order we clear insurance pools
+	// changes the division when we split it across remaining markets and
+	// who the remainder ends up with if it doesn't divide equally.
+	allIDs := []string{}
```

Comment:
it's probably not a big deal now because it's a small number but maybe it's worth keeping a marketCPStatesCpy as a slice because this is called on tick.